### PR TITLE
Add section on functions

### PR DESCRIPTION
**Pull request by anderseknert**
Signed-off-by: Anders Eknert <anders@eknert.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -730,6 +733,64 @@ allow {
...
+
+**Prefer**
+```rego
+first_a := i {
+    i := indexof("answer", "a")
+}
+```
+
+While the second form is valid, it is almost guaranteed to confuse developers coming from the most common programming
+languages, which is likely most developers. Again, optimize for readability!
```
Comment (tsandall) [inline README.md]: "which is likely most developers" seems unnecessary.

File: README.md [hunk 2]
```diff
@@ -730,6 +733,64 @@ allow {
 #### Related Resources
 - [Five things you didn't know about OPA](https://blog.styra.com/blog/five-things-you-didnt-know-about-opa).
 
+## Functions
+
+### Prefer using arguments over `input` and `data`
+
+Rules may freely reference values from `input` and `data`. What separates functions from rules is that they accept
+_arguments_. While functions too may reference anything from `input` and `data`, these references create dependencies
+that aren't obvious simply by checking the function signature, and it makes it much harde to reuse a function like that
```
Comment (tsandall) [inline README.md]: s/harde/harder/

Comment (anderseknert) [reply README.md]: Thanks! Fixed.

File: README.md (hunk 1 above)
Comment (anderseknert) [reply README.md]: 👍 Removed

**Pull request merged**